ydb/core/tx/schemeshard/schemeshard_info_types.cpp

```diff
@@ -1436,6 +1575,30 @@ void TAggregatedStats::UpdateShardStats(TShardIdx datashardIdx, const TPartition
     Aggregated.RowCount += (newStats.RowCount - oldStats.RowCount);
     Aggregated.DataSize += (newStats.DataSize - oldStats.DataSize);
     Aggregated.IndexSize += (newStats.IndexSize - oldStats.IndexSize);
+    for (const auto& [poolKind, newStoragePoolStats] : newStats.StoragePoolsStats) {
+        auto* aggregatedStoragePoolStats = Aggregated.StoragePoolsStats.FindPtr(poolKind);
+        if (aggregatedStoragePoolStats) {
+            const auto* oldStoragePoolStats = oldStats.StoragePoolsStats.FindPtr(poolKind);
+
+            // Missing old stats for a particular storage pool are interpreted as if this data
+            // has just been written to the datashard and we need to increment the aggregate by the entire new stats' sizes.
+            aggregatedStoragePoolStats->DataSize += newStoragePoolStats.DataSize - (oldStoragePoolStats ? oldStoragePoolStats->DataSize : 0u);
+            aggregatedStoragePoolStats->IndexSize += newStoragePoolStats.IndexSize - (oldStoragePoolStats ? oldStoragePoolStats->IndexSize : 0u);
+        } else {
+            Aggregated.StoragePoolsStats.emplace(poolKind, newStoragePoolStats);
+        }
+    }
+    for (const auto& [poolKind, oldStoragePoolStats] : oldStats.StoragePoolsStats) {
+        if (const auto* newStoragePoolStats = newStats.StoragePoolsStats.FindPtr(poolKind); !newStoragePoolStats) {
+            auto* aggregatedStoragePoolStats = Aggregated.StoragePoolsStats.FindPtr(poolKind);
+            Y_ENSURE(aggregatedStoragePoolStats); // How else could it be present in the oldStats?
```

Comment:
@ijon Is it ok to potentially throw an error here? (I think this condition should never be violated.)